src/cfnlint/core.py

```diff
@@ -0,0 +1,365 @@
+                if ignore_bad_template:
+                    LOGGER.info('Template %s is malformed: %s', filename, err.problem)
+                    LOGGER.error('Tried to parse %s as JSON but got error: %s', filename, str(json_err))
+                else:
+                    LOGGER.error('Template %s is malformed: %s', filename, err.problem)
+                    LOGGER.error('Tried to parse %s as JSON but got error: %s', filename, str(json_err))
+                sys.exit(1)
+            except Exception as json_err:  # pylint: disable=W0703
+                if ignore_bad_template:
+                    LOGGER.info('Template %s is malformed: %s', filename, err.problem)
+                    LOGGER.info('Tried to parse %s as JSON but got error: %s', filename, str(json_err))
+                else:
+                    LOGGER.error('Template %s is malformed: %s', filename, err.problem)
+                    LOGGER.error('Tried to parse %s as JSON but got error: %s', filename, str(json_err))
+                    sys.exit(1)
+
+    append_parser(parser, defaults)
+    args = parser.parse_args(cli_args)
+
+    if vars(args)['update_specs']:
+        cfnlint.helpers.update_resource_specs()
+        exit(0)
+
+    rules = cfnlint.core.get_rules(vars(args)['append_rules'])
+
+    if vars(args)['listrules']:
+        print(rules)
+        exit(0)
+
+    if not vars(args)['template']:
+        parser.print_help()
+        exit(1)
+
+    return(args, template, rules)
+
+
+def get_default_args(template):
+    """ Parse and validate default args """
+    defaults = {}
+    if isinstance(template, dict):
+        configs = template.get('Metadata', {}).get('cfn-lint', {}).get('config', {})
+
+    if isinstance(configs, dict):
+        for config_name, config_value in configs.items():
+            if config_name == 'ignore_checks':
+                if isinstance(config_value, list):
+                    defaults['ignore_checks'] = config_value
+            if config_name == 'regions':
+                if isinstance(config_value, list):
+                    defaults['regions'] = config_value
+            if config_name == 'append_rules':
+                if isinstance(config_value, list):
+                    defaults['override_spec'] = config_value
+            if config_name == 'override_spec':
+                if isinstance(config_value, (six.string_types, six.text_type)):
+                    defaults['override_spec'] = config_value
+            if config_name == 'ignore_bad_template':
+                if isinstance(config_value, bool):
+                    defaults['ignore_bad_template'] = config_value
+            if config_name == 'log_level':
+                if isinstance(config_value, (six.string_types, six.text_type)):
+                    defaults['log_level'] = config_value
+
+    return defaults
+
+
+def run_checks(filename, template, rules, transforms, ignore_checks, regions):
+    """Run Checks against the template"""
+    if regions:
```

Comment:
Does it make sense to break out supported regions into a constant?  That way we can re-use elsewhere and easily update if more regions are added.

Reply:
Most recent change should cover this.  We were also setting a Regions variable in helpers and so I removed this one and used that one.

Reply:
Ha! No wonder I had the idea to break them out in my head -- we already had! :)  Looks good.

👍